lib/msf/core/auxiliary/report.rb

```diff
@@ -57,6 +57,7 @@ def create_credential_and_login(opts={})
       framework.db.create_credential_and_login(opts)
     elsif !db_warning_given?
       vprint_warning('No active DB -- Credential data will not be saved!')
+      nil
```

Comment:
This change stops the following issue:

Module code:
```
credential_login = create_credential_and_login(credential_data)
credential_login&.core_id
```

Library code:

https://github.com/rapid7/metasploit-framework/blob/c218063a1aca2545fa4071fefc5be4198c17fe37/lib/msf/core/auxiliary/report.rb#L54-L61

Error:
```
[-] 192.168.123.13:445 - Auxiliary failed: NoMethodError undefined method `core_id' for "\e[1m\e[33m[!]\e[0m 192.168.123.13:445 - No active DB -- Credential data will not be saved!\n":String
```

The crash only happens when verbose mode is enabled and there's no DB attached. It should be applied to the other methods too, but that will be done separately